pjsip/src/pjsip/sip_transport_tcp.c

```diff
@@ -1537,6 +1537,12 @@ static pj_bool_t on_connect_complete(pj_activesock_t *asock,
 
 	tcp_perror(tcp->base.obj_name, "TCP connect() error", status);
 
+  /* Mark the transport as failed, ahead of it being shutdown below.  This
+   * avoids it from being selected to send any further messages as a result
+   * of processing in the on_data_sent(), which happens before the transaction
+   * gets marked as shutdown. */
+  tcp->base.is_failed = PJ_TRUE;
```

Comment:
Indentation.  I believe this file uses tabs rather than spaces (because it's PJSIP's, not Clearwater's).